apps/aecore/lib/aecore/peers/worker.ex

```diff
@@ -109,17 +123,28 @@ defmodule Aecore.Peers.Worker do
     {:reply, peers, peers}
   end
 
-  def handle_cast({:broadcast_tx, tx}, peers) do
-    serialized_tx = 
-    Serialization.tx(tx, :serialize)
-    |> Poison.encode!()
-    for peer <- peers do
-      Client.send_tx(peer, serialized_tx)
-    end
+  ## Async operations
 
+  def handle_cast({:async_request, {type, data}}, peers) do
+    send_to_peers(:post, type, data, Map.keys(peers))
     {:noreply, peers}
   end
+
   def handle_cast(_any, peers) do
-    {:noreply, peers}
+    Logger.info("[Peers] Unhandled cast message:  #{inspect(peers)}")
+  end
+
+  ## Internal functions
+  defp send_to_peers(:post, uri, data, []) do
```

Comment:
there sould be no spectial case, `for peer <- peers do` should work with empty list. You can log to debug after sending to how many peers it was sent